plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -0,0 +1,244 @@
+          _title = mkCodeActionTitle exts crs
+        , _kind = Just CodeActionRefactorRewrite
+        , _diagnostics = Nothing
+        , _isPreferred = Nothing
+        , _disabled = Nothing
+        , _edit = Just $ WorkspaceEdit (changes crs) Nothing Nothing
+        , _command = Nothing
+        , _xdata = Nothing
+        }
+    actions = map mkCodeAction (RangeMap.filterByRange caRange crsMap)
+  pure $ List actions
+  where
+    mkCodeActionTitle :: [Extension] -> RecordSelectorExpr-> Text
+    mkCodeActionTitle exts (RecordSelectorExpr _ se _) =
+      if OverloadedRecordDot `elem` exts
+        then title
+        else title <> " (needs extension: OverloadedRecordDot)"
+        where
+          title = "Convert `" <> name <> "` to record dot syntax"
+          name = printOutputable se
+
+collectRecSelsRule :: Recorder (WithPriority Log) -> Rules ()
+collectRecSelsRule recorder = define (cmapWithPrio LogShake recorder) $ \CollectRecordSelectors nfp ->
+  useWithStale TypeCheck nfp >>= \case
+    Nothing -> pure ([], Nothing)
+    Just (tmr, pm) -> do
+      let exts = getEnabledExtensions tmr -- We need the extensions to check whether we need to add OverloadedRecordDot
+          recSels = mapMaybe (rewriteRange pm) (getRecordSelectors tmr) -- And we need the record selectors for obvious reasons
+      logWith recorder Debug (LogCollectedRecordSelectors recSels)
+      let crsMap :: RangeMap RecordSelectorExpr
+          crsMap = RangeMap.fromList location recSels -- We need the rangeMap to be able to filter by range later
+      pure ([], CRSR <$> Just crsMap <*> Just exts)
+  where
+    getEnabledExtensions :: TcModuleResult -> [Extension]
+    getEnabledExtensions = getExtensions . tmrParsed
+    getRecordSelectors :: TcModuleResult -> [RecordSelectorExpr]
+    getRecordSelectors (tmrRenamed -> (hs_valds -> valBinds,_,_,_)) =
+        collectRecordSelectors valBinds
+    rewriteRange :: PositionMapping -> RecordSelectorExpr -> Maybe RecordSelectorExpr
+    rewriteRange pm recSel = case toCurrentRange pm (location recSel) of
+                                Just newLoc -> Just $ recSel{location = newLoc}
+                                Nothing     -> Nothing
+
+convertRecordSelectors :: RecordSelectorExpr ->  TextEdit
+convertRecordSelectors (RecordSelectorExpr l se re) = TextEdit l $ convertRecSel se re
+
+-- | Converts a record selector expression into record dot syntax,
+-- | currently we are using printOutputable to do it. We are also letting GHC
+-- | decide when to parenthesize the record expression
+convertRecSel :: Outputable (LHsExpr (GhcPass 'Renamed)) => LHsExpr (GhcPass 'Renamed) -> LHsExpr (GhcPass 'Renamed) -> Text
+convertRecSel se re = printOutputable (parenthesizeHsExpr appPrec re) <> "." <> printOutputable se
+
+-- It's important that we use everthingBut here, because if we used everything we would
+-- get duplicates for every case that occurs inside a HsExpanded expression.
```

Comment:
worth writing an example? You gave one in the issue, it seems a shame to lose it. Maybe a test case?

Reply:
Yes. Wrote a test and referred to it from the comments.